config/yaml.go

```diff
@@ -76,40 +112,29 @@ func NewYAMLProviderFromFiles(mustExist bool, resolver FileResolver, files ...st
 	return newYAMLProviderCore(readers...)
 }
 
-// NewYamlProviderFromReader creates a configuration provider from an
-// io.ReadCloser
-func NewYamlProviderFromReader(reader io.ReadCloser) Provider {
-	return newYAMLProviderCore(reader)
+// NewYamlProviderFromReader creates a configuration provider from an io.ReadClosers.
```

Comment:
nit: `an` suggests a single `ReadCloser`, but you take multiple. maybe:

```
creates a configuration provider from a list of `ReadClosers`
```
or something to indicate plurality